tests/Unit/ControlSystem/ControlErrors/Test_SizeError.cpp

```diff
@@ -161,6 +165,25 @@ void test_size_error(const double grid_excision_boundary_radius,
       target_char_speed, function_of_time, expected_error);
 }
 
+void test_size_error_class() {
+  using size_error =
+      control_system::ControlErrors::Size<domain::ObjectLabel::A>;
+  static_assert(
+      tt::assert_conforms_to_v<size_error,
+                               control_system::protocols::ControlError>);
+
+  auto error = TestHelpers::test_creation<size_error>(
+      "MaxNumTimesForZeroCrossingPredictor: 4");
+
+  CHECK_FALSE(error.get_suggested_timescale().has_value());
+  CHECK_FALSE(error.discontinuous_change_has_occurred());
+
+  error.reset();
+
+  CHECK_FALSE(error.get_suggested_timescale().has_value());
+  CHECK_FALSE(error.discontinuous_change_has_occurred());
```

Comment:
Is it possible to test a situation where the class actually does something?

Reply:
Technically yes, but it'd be a lot of work. I'd have to set up a cache and all the measurements with very specific values so that 1) I don't get an FPE and 2) the states actually suggest changing to a different state so you get a discontinuous change. All these combinations for switching states are tested in the `control_system::size::control_error` function, which this class calls, which is why I didn't copy that work here.

Reply:
If it's not simple you don't need to test a state change, since that's done elsewhere, but `operator()` is about half the code in the class.  You should at least call it, even if you can only do rough sanity checking on the result.